### PR TITLE
Create manifest JSON file with mapping to initial assets

### PR DESCRIPTION
**Pull request by ronami**
### 🔦 Summary

This PR adds on `build-app` to create an assets manifest JSON file. The manifest should be used by tools and platforms to get the full mappings of all initial assets, per entry.

Currently, the manifest file will be created only on CI and will include the version in its name (and without any hash). That will allow tools and platforms to get the mappings for a specific version.

### CODE REVIEW

File: packages/yoshi-helpers/utils.js [hunk 1]
```diff
@@ -250,7 +256,7 @@ module.exports.killSpawnProcessAndHisChildren = child => {
 
     const pid = child.pid;
 
-    psTree(pid, (err /*eslint handle-callback-err: 0*/, children) => {
+    psTree(pid, (err, children) => {
```
Comment (ronami) [inline packages/yoshi-helpers/utils.js]: why did we have this lint override? 😅

Comment (ronami) [inline packages/yoshi-helpers/utils.js]: why did we have this lint override? 😅

File: packages/yoshi/src/commands/build-app.js [hunk 2]
```diff
@@ -128,6 +131,44 @@ module.exports = async () => {
 
   const clientOptimizedStats = webpackStats.stats[1];
 
+  // Generate `manifest.[version].json`
+  if (inTeamCity && experimentalBuildHtml) {
+    const assetsJson = clientOptimizedStats.compilation.chunkGroups.reduce(
```
Comment (netanelgilad) [inline packages/yoshi/src/commands/build-app.js]: This piece of code seems a bit complicated to grok. What about some tests? even units could be nice.

File: packages/yoshi/src/commands/build-app.js [hunk 3]
```diff
@@ -128,6 +131,56 @@ module.exports = async () => {
...
+          ...(acc[chunk.name] || []),
+          // Add files to the list
+          ...chunk.chunks.reduce(
+            (files, child) => [
+              ...files,
+              ...child.files
+                // Remove map files
+                .filter(file => !file.endsWith('.map'))
+                // Remove rtl.min.css files
+                .filter(file => !file.endsWith('.rtl.min.css'))
```
Comment (netanelgilad) [inline packages/yoshi/src/commands/build-app.js]: not needed

**Pull request merged**